scripts/list_hardware.py

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+from dataclasses import dataclass
+from pathlib import Path, PurePath
+import pykwalify.core
+import sys
+from typing import List
+import yaml
+
+
+SOC_SCHEMA_PATH = str(Path(__file__).parent / 'schemas' / 'soc-schema.yml')
+with open(SOC_SCHEMA_PATH, 'r') as f:
+    soc_schema = yaml.safe_load(f.read())
+
+ARCH_METADATA_SCHEMA = '''
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The archs.yml file is a simple list of key value pairs containing architectures
+# and their location which is used by the build system.
+type: map
+mapping:
```

Comment:
suggestion: store schema in a separate file, same as soc-schema.yml

Reply:
done